GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3083,7 +3083,17 @@ private void RevisionInfo_CommandClick(object sender, CommitInfo.CommandEventArg
         {
             if (e.Command == "gotocommit")
             {
-                RevisionGrid.SetSelectedRevision(new GitRevision(Module, e.Data));
+                var revision = new GitRevision(Module, e.Data);
+                var found = RevisionGrid.SetSelectedRevision(revision);
+
+                // When 'git log --first-parent' filtration is used, user can click on child commit
+                // that is not present in the shown git log. User still wants to see the child commit
+                // and to make it possible we add explicit branch filter and refresh.
+                if (AppSettings.ShowFirstParent && !found)
+                {
+                    _filterBranchHelper.SetBranchFilter(e.Data, refresh: true);
```

Comment:
Yes, please.
`_filterBranchHelper.SetBranchFilter(revision.Guid, refresh: true);`
 is more descriptive than
`_filterBranchHelper.SetBranchFilter(e.Data, refresh: true);`